src/core/jsproxy.c

```diff
@@ -222,13 +259,84 @@ JsProxy_length(PyObject* o)
   return hiwire_get_length(self->js);
 }
 
+static PyObject*
+JsProxy_subscript_array(PyObject* o, PyObject* item)
+{
+  JsProxy* self = (JsProxy*)o;
+  if (PyIndex_Check(item)) {
+    Py_ssize_t i;
+    i = PyNumber_AsSsize_t(item, PyExc_IndexError);
+    if (i == -1 && PyErr_Occurred())
+      return NULL;
+    if (i < 0)
+      i += hiwire_get_length(self->js);
+    JsRef result = hiwire_get_member_int(self->js, i);
+    if (result == NULL) {
+      PyErr_SetObject(PyExc_KeyError, item);
+      return NULL;
+    }
+    PyObject* pyresult = js2python(result);
+    hiwire_decref(result);
+    return pyresult;
+  }
+  if (PySlice_Check(item)) {
+    PyErr_SetString(PyExc_NotImplementedError,
+                    "Haven't implemented slice subscript yet");
+    return NULL;
+  }
+  PyErr_Format(PyExc_TypeError,
+               "list indices must be integers or slices, not %.200s",
```

Comment:
This is copied verbatim from the CPython source. I don't know what it means either. Note that Python doesn't use the standard `printf` formatters, it uses [`PyUnicode_FromFormat`](https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_FromFormat) which is similar but slightly different in not fully documented ways.

Here is where I copied this from:
https://github.com/python/cpython/blob/213c155a460b8dd9e43901e4d61aa088cbac4221/Objects/listobject.c#L2856

Reply:
Presumably this truncates the length of the string to 200 if you have an extremely long type name?

Reply:
I found the relevant CPython comment:

>      *** WARNING ***
>
>       The format should limit the total size of the formatted output string to
>       1000 bytes.  In particular, this means that no unrestricted "%s" formats
>       should occur; these should be limited using "%.<N>s where <N> is a
>       decimal number calculated so that <N> plus the maximum size of other
>       formatted text does not exceed 1000 bytes.  Also watch out for "%f",
>       which can print hundreds of digits for very large numbers.

https://github.com/python/cpython/blob/e89380765df8f0f02c90ad417e164d1597bd0b05/Python/sysmodule.c#L3243